### DISCUSSION TITLE
Multi-Store Option

### DISCUSSION THREAD

**Discussion by ch3ckmat3** [medusajs/medusa#2142] [Q&A] [answered]
Since Medusa is being dubbed as Shopify a alternative, I expect it to offer multi-store features, where multiple store owners can open shops on a single installation of Medusa (like Amazon or AliExpress) and buyers are free to shop from any store via single account. Is this feature available in any way, or planned for the future?

**Comment by mikemoooo** [comment]
This would be something that is necessary for us.

**Comment by olivermrbl** [comment, marked as answer]
From what I can understand, you are both looking for marketplace functionality, which is not supported natively in Medusa. 

Though, we've helped dozens of merchants achieve such a setup which has led to [this blog series](https://medusajs.com/blog/medusa-extender). Essentially, you will need to use an extender tool created by one of our core team members allowing you to extend the database schema of your Medusa instance.

I'll let you check out the tutorial. Please don't hesitate to ask follow-up questions here, and we'll try our best to help out.

**Reply by mariusa** [reply]
Hi, this isn't about marketplace. As the original poster said, it would allow one Medusa instance (within a single DB schema) to host multiple merchants, each with their own store, customers, orders etc. Just like Shopify does. 
The longer you wait for adding this, the harder is becomes. We'd need say an "org" table (for each merchant/business) and an org_id column added to all or some of the existing tables, with queries modified to filter by the current org_id too. 

A regular setup for just one merchant would have just one org, there's no downside to that. There's a huge downside to making everybody who needs this go through https://medusajs.com/blog/medusa-extender and then finding other issues in time, due to not being supported officially.

Thanks

**Reply by gisellfaithbillones** [reply]
> Hi, this isn't about marketplace. As the original poster said, it would allow one Medusa instance (within a single DB schema) to host multiple merchants, each with their own store, customers, orders etc. Just like Shopify does. The longer you wait for adding this, the harder is becomes. We'd need say an "org" table (for each merchant/business) and an org_id column added to all or some of the existing tables, with queries modified to filter by the current org_id too.
> 
> A regular setup for just one merchant would have just one org, there's no downside to that. There's a huge downside to making everybody who needs this go through https://medusajs.com/blog/medusa-extender and then finding other issues in time, due to not being supported officially.
> 
> Thanks

Same question, is this feature available now?

**Comment by jaruesink** [comment]
I don't believe Shopify offers multi-store functionality out of the box. They require 3rd party plugins that are not great. However, we are building marketplace functionality on top of Medusa and are happy to work with teams who need that functionality. Thanks!

**Reply by NitishGameChanges** [reply]
Hi, We have added multiple fulfillment providers by extending FulfilmentService but based on the vendor we want to have a set of providers only for a vendor. How can this be achievable? @jaruesink @olivermrbl

**Reply by jaruesink** [reply]
Sounds like you'd have to make a way to select which subset of fulfillment providers you want associated with each vendor and filter based on that. It's not something baked into Medusa, so it would require some new architecture.

**Reply by NitishGameChanges** [reply]
How about making shipping options associated with each vendor and filtering on that? Only when shipping option is added for a vendor it will be shown to that vendor.

**Reply by dwene** [reply]
We used the medusa-extender to overhaul the fulfillment-provider service in medusa, and we extended the shipping-profile database table to have a vendor_id, so each vendor could create their own shipping options. It was quite a bit of work though, doing that in the admin UI and api.

**Reply by mariusa** [reply]
Hi, this isn't about marketplace. As the original poster said, it would allow one Medusa instance (within a single DB schema) to host multiple merchants, each with their own store, customers, orders etc. Just like Shopify does. At shopify.com, many merchants can signup to have their own shop.

The longer you wait for adding this, the harder is becomes. We'd need say an "org" table (for each merchant/business) and an org_id column added to all or some of the existing tables, with queries modified to filter by the current org_id too.

A regular setup for just one merchant would have just one org, there's no downside to that. There's a huge downside to making everybody who needs this go through https://medusajs.com/blog/medusa-extender and then finding other issues in time, due to not being supported officially (see @dwene 's comment). 

Thanks

**Comment by NitishGameChanges** [comment]
What about having multiple payment providers and mapping them with vendors? So a vendor will have a subset of payment_providers.  Which table and method should be extended/overridden in order to achieve it or what alternative approaches are possible?  @jaruesink @dwene @olivermrbl

**Reply by intrinsic-blockchain** [reply]
This is exactly what I thought when I saw that the payment provider API keys are saved in environment files in the official plugins when there should have been through a UI may be in the Admin UI somewhere separate for each store. I am not sure we can change that when using the official plugins. May be need to write own plugins to support that.

**Reply by WassilaBeny** [reply]
Did anyone find a solution for that? @intrinsic-blockchain @NitishGameChanges

**Comment by danvayn** [comment]
Just bumping this thread with the relevant page since no one else has shared it:
https://docs.medusajs.com/recipes/marketplace

**Reply by lacymorrow** [reply]
It isn't complete, as there is no `members` property created in new stores
